### DISCUSSION TITLE
https://thanhle.blog/blog/tuong-lai-cua-frontend

### DISCUSSION THREAD

**Discussion by giscus** [thanhlmm/blog#26] [Ideas]
# https://thanhle.blog/blog/tuong-lai-cua-frontend

Viễn cảnh tương lai của frontend

https://thanhle.blog/blog/tuong-lai-cua-frontend

**Comment by nguyenhoaikhang37** [comment]
Bài viết quá thú vị luôn anh Thành, quá khứu front-end chỉ cắt html/css là có đc job đủ sống, tương lai chắc phải thành nồi lẩu thập cẩm mới có lẩu ăn :))) Trong bài em thấy có nói tới thằng TS, anh có tính làm bài viết về viết TS sao trong app react cho hiệu quá k ạ, em đang bị case là cài mấy thư viện như react-query, thì k phải PropType nào cũng biết (nên em để any luôn, mặc dù biết là k nên)

**Reply by thanhlmm** [reply]
TypeScript thì anh cũng tay ngang thôi, đang định có time sẽ học lại bài bản. Dùng any theo anh cũng không có gì sai cả, nhưng nên hạn chế nhất có thể thôi. Việc làm deliver đúng deadline thì quan trọng hơn rất nhiều time để mò đúng cái type :D

**Comment by oucommunity** [comment]
em thấy anh nhúng giscus này vào web, em cũng thử tìm hiểu nhúng giscus này vào github page nhưng nó hiện lỗi như dưới. Anh có cách khắc phục vấn đề này k ạ ? Em cảm ơn
https://user-images.githubusercontent.com/106130487/171262023-a1a18518-8354-4e1b-8fc8-961fe1cdf166.png

**Reply by thanhlmm** [reply]
Có thể Github Page chặn nhúng iframe đó em nên em thử host chỗ khác xem. Anh đang host ở Vercel.

**Reply by thanhlmm** [reply]
BTW nếu em mở console trang web này sẽ thấy lỗi y chang. Nên chắc không phải do lỗi đó đâu, em coi lại phần integrate xem đúng chưa?

**Comment by darkpiv** [comment]
>Mình tin là tương lại blockchain sẽ nhẹ hơn để có thể run được ở browser. Solve được bài toàn Decentralized hoàn toàn khi ai cũng có thể tham gia vào vận hành blockchain
Hi vọng là cái này dc solved sớm sẽ unlock nhiều thứ :D 
Btw ko biết tương lai device mn dự đoán nó sẽ ntn hen? Sau bùng nổ của smartphone, smartwatch (chưa thực sự bùng nổ) thì ko biết next sẽ là gì? smart-XXX

**Reply by darkpiv** [reply]
ủa ko edit được =)))

**Reply by thanhlmm** [reply]
Theo e device chắc chưa có gì thay đổi nhiều sau 5 năm đâu. Next có thể là VR nhưng chắc cũng phải tầm 10 năm :D

**Comment by thanhlmm** [comment]
Figma vừa được Adobe mua lại với định giá 2 tỷ đô. Đương nhiên sẽ có nhiều lý do để đưa tới định giá và quyết định mua của Adobe.
Tuy nhiên điều mình thấy Figma có sự khác bọt mà Adobe mong muốn có được, là một ứng dụng 100% web cho designer. Điều mà Adobe XD, Photoshop không làm được hoặc không thể làm đủ tốt.

> Almost everything is web

**Comment by beocaca** [comment]
bài viết hay qúa anh ơi, tìm được chiếc blog đỉnk gheeee

**Reply by thanhlmm** [reply]
Thanks em. Nhớ subcribe email để nhận bài viết mỗi tuần nhé ^^

**Comment by vietanhdev** [comment]
Đến thời điểm này chắc có thêm các ảnh hưởng của AI nữa anh nhỉ?

**Reply by thanhlmm** [reply]
yes, đùng là thời điểm viết bài mình ko biết j tời AI. Tới giờ thì...
